src/Http/Controllers/SignedStorageUrlController.php

```diff
@@ -118,6 +118,7 @@ protected function storageClient()
     {
         $config = [
             'region' => $_ENV['AWS_DEFAULT_REGION'],
+            'url' => $_ENV['AWS_URL'] ?? null,
```

Comment:
@themsaid I think the parameter is `endpoint` (https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_configuration.html#endpoint)

Reply:
Do you know why it's set as `url` in laravel/laravel then?

Reply:
@clmntgr did you test that `endpoint` using minio?

Reply:
> Do you know why it's set as `url` in laravel/laravel then?

No idea, but it was said in https://github.com/laravel/laravel/pull/4483 too

> @clmntgr did you test that `endpoint` using minio?

No, I will test it today

Reply:
@themsaid I was not able to make it work with minio but succeed with [localstack](https://github.com/localstack/localstack).

But some more changes are required, [here](https://github.com/laravel/vapor-core/pull/20/files#diff-2b0e794964ec76866c75e3a1f408569aR45) you need to directly pass  `(string) $uri`, as the current code will remove the port from `AWS_URL`. Is there a reason why this code is like that?

You also need to add `'use_path_style_endpoint' => true,` in the config send to AWS because of [this](https://github.com/localstack/localstack#troubleshooting). There is an issue to improve this (https://github.com/localstack/localstack/issues/1792) but not sure its possible, maybe only add this option if `AWS_URL` target localhost?